### PR TITLE
[ELF] Rename LazyObject to LazySymbol. NFC

### PR DESCRIPTION
**Pull request by MaskRay**
LazySymbol (used by wasm port) is a more accurate name (the struct is
not about an object). However, the ELF port calls this LazyObject likely
because we used to have LazyArchive (removed by
https://reviews.llvm.org/D119074), and LazyObject has a similar naming
convention (LazyObjectSymbol would be better, but it is too long).

### CODE REVIEW

**Comment by sbc100** [comment]
Awesome!  Thanks.   I just created that same change myself but now I don't need to upload it.

File: lld/ELF/Symbols.cpp [hunk 1]
```diff
@@ -146,7 +146,7 @@ static uint64_t getSymVA(const Symbol &sym, int64_t addend) {
   case Symbol::SharedKind:
   case Symbol::UndefinedKind:
     return 0;
-  case Symbol::LazyObjectKind:
+  case Symbol::LazySymbolKind:
```
Comment (sbc100) [inline lld/ELF/Symbols.cpp:149]: I think this can also just be `LazyKind` since its matches the other kinds.

**Comment by sbc100** [comment]
(Should the title prefix be `[lld][ELF]`?)

File: lld/ELF/Symbols.cpp (hunk 1 above)
Comment (MaskRay) [reply lld/ELF/Symbols.cpp]: Thanks for the suggestion!

**Comment by MaskRay** [comment]
> (Should the title prefix be `[lld][ELF]`?)

I follow George Rimar and some others' convention by using just `[ELF]`. It's like for many llvm and clang changes, many people don't use `[llvm]` or `[clang]`. This makes sense, because you need `git log -- llvm` to figure out llvm/ changes to not bother with other changes, and the `[llvm]` tag is not very helpful when I run git log this way. I think the `XXX` in `llvm/lib/XXX` `clang/lib/XXX` part is more useful. However, I see that many people add the XXX in `llvm-project/XXX` as a tag. I think it's different preference.

**Pull request merged**